File: frontend/src/types/place.ts
export interface Place {
  id: number;
  name: string;
  address: string | null;
  postcode: string | null;
  sub_area: string | null;
  area: string | null;
  state: string | null;
  category: string | null;
  seating: string | null;
  remarks: string | null;
  latitude: number | null;
  longitude: number | null;
}

export interface PlaceListResponse {
  total: number;
  items: Place[];
}

export interface Filters {
  category: string;
  seating: string;
  state: string;
}

export interface Submission {
  id: number;
  name: string;
  address: string;
  state: string | null;
  category: string | null;
  seating: string | null;
  remarks: string | null;
  google_maps_url: string | null;
  status: "pending" | "approved" | "rejected";
  submitted_by: string | null;
  reviewed_by: string | null;
  created_at: string;
}

File: backend/app/schemas/place.py
from pydantic import BaseModel, ConfigDict


class PlaceBase(BaseModel):
    name: str
    address: str | None = None
    postcode: str | None = None
    sub_area: str | None = None
    area: str | None = None
    state: str | None = None
    category: str | None = None
    seating: str | None = None
    remarks: str | None = None
    latitude: float | None = None
    longitude: float | None = None


class PlaceUpdate(BaseModel):
    name: str | None = None
    address: str | None = None
    postcode: str | None = None
    sub_area: str | None = None
    area: str | None = None
    state: str | None = None
    category: str | None = None
    seating: str | None = None
    remarks: str | None = None
    latitude: float | None = None
    longitude: float | None = None


class PlaceOut(PlaceBase):
    model_config = ConfigDict(from_attributes=True)

    id: int


class PlaceListOut(BaseModel):
    total: int
    items: list[PlaceOut]


class StateOut(BaseModel):
    state: str


class AreaOut(BaseModel):
    area: str
    sub_areas: list[str]
